static/js/lib/api/courses.js

```diff
@@ -1,7 +1,66 @@
 // @flow
+import R from "ramda"
+
 import { fetchJSONWithAuthFailure } from "./fetch_auth"
+
 import type { Course } from "../../flow/discussionTypes"
 
 export function getCourse(courseId: number): Promise<Course> {
   return fetchJSONWithAuthFailure(`/api/v0/courses/${courseId}/`)
 }
+
+export const featuredCoursesSelector = R.pathOr(null, [
+  "entities",
+  "featuredCourses"
+])
+export const featuredCoursesRequest = () => ({
+  url:       "/api/v0/courses/featured",
+  transform: (responseJson: Object) => {
+    const { next, results } = responseJson
+
+    return {
+      featuredCourses:     results,
+      featuredCoursesNext: next
+    }
+  },
+  update: {
+    featuredCourses: (prev: Array<Course>, next: Array<Course>) => next
+  }
+})
+
+export const upcomingCoursesSelector = R.pathOr(null, [
+  "entities",
+  "upcomingCourses"
+])
+export const upcomingCoursesRequest = () => ({
+  url:       "/api/v0/courses/upcoming",
+  transform: (responseJson: Object) => {
+    const { next, results } = responseJson
+
+    return {
+      upcomingCourses:    results,
+      upcomingCourseNext: next
+    }
+  },
+  update: {
+    upcomingCourses:    (prev: Array<Course>, next: Array<Course>) => next,
+    upcomingCourseNext: (prev: string, next: string) => next
+  }
+})
+
+export const newCoursesSelector = R.pathOr(null, ["entities", "newCourses"])
```

Comment:
Looks like there are some test coverage gaps here according to codecov, maybe add unit tests for `newCoursesRequest`, `upcomingCoursesRequest` and `featuredCoursesRequest`

Reply:
sure, I left them untested because they're basically boilerplate, but I can add a few tests so we at least have regression tests